File: v1/src/main/java/com/api/v1/payment/PaymentResponseMapper.java
package com.api.v1.payment;

import com.api.v1.car.domain.Car;
import com.api.v1.car.mappers.CarResponseMapper;
import com.api.v1.creditcar.domain.CreditCard;
import com.api.v1.creditcar.mappers.CreditCardResponseMapper;
import com.api.v1.customer.domain.Customer;
import com.api.v1.customer.mappers.CustomerResponseMapper;

public record PaymentResponseMapper(
        Customer customer,
        Car car,
        CreditCard creditCard,
        Payment payment
) {

    PaymentResponseDto map() {
        return new PaymentResponseDto(
                new CustomerResponseMapper(customer).mapToDto(),
                new CarResponseMapper(car).mapToDto(),
                new CreditCardResponseMapper(creditCard).mapToDto(),
                payment.getPaidAt()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/payment/mappers/PaymentResponseMapper.kt
package com.api.v2.payment.mappers

import com.api.v2.car.domain.Car
import com.api.v2.car.mappers.CarResponseMapper
import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.creditcard.mappers.CreditCardResponseMapper
import com.api.v2.customer.domain.Customer
import com.api.v2.customer.mappers.CustomerResponseMapper
import com.api.v2.payment.domain.Payment
import com.api.v2.payment.dtos.PaymentResponseDto
import java.time.ZonedDateTime

class PaymentResponseMapper {

    companion object {
        fun map(customer: Customer, car: Car, creditCard: CreditCard): PaymentResponseDto {
            return PaymentResponseDto(
                CustomerResponseMapper.mapToDto(customer),
                CarResponseMapper.mapToDto(car),
                CreditCardResponseMapper.mapToDto(creditCard),
                ZonedDateTime.now().toString()
            )
        }
    }

}